routes/api/users.js

```diff
@@ -1,87 +1,105 @@
+        user.username = req.body.user.username;
+      }
+      if (typeof req.body.user.email !== 'undefined') {
+        user.email = req.body.user.email;
+      }
+      if (typeof req.body.user.bio !== 'undefined') {
+        user.bio = req.body.user.bio;
+      }
+      if (typeof req.body.user.image !== 'undefined') {
+        user.image = req.body.user.image;
+      }
+      if (typeof req.body.user.password !== 'undefined') {
+        user.setPassword(req.body.user.password);
+      }
 
-            return user.save().then(function() {
-                return res.json({ user: user.toAuthJSON() });
-            });
-        })
-        .catch(next);
+      return user.save().then(() => res.json({ user: user.toAuthJSON() }));
+    })
+    .catch(next);
 });
 
-router.post("/users/login", function(req, res, next) {
-    if (!req.body.user.email) {
-        return res.status(422).json({ errors: { email: "can't be blank" } });
-    }
+router.post('/users/login', (req, res, next) => {
+  if (!req.body.user.email) {
+    return res.status(422).json({ errors: { email: "can't be blank" } });
+  }
 
-    if (!req.body.user.password) {
-        return res.status(422).json({ errors: { password: "can't be blank" } });
+  if (!req.body.user.password) {
+    return res.status(422).json({ errors: { password: "can't be blank" } });
+  }
+  passport.authenticate('local', { session: false }, (
+    err,
+    user,
+    info
+  ) => {
+    if (err) {
+      return next(err);
     }
-    passport.authenticate("local", { session: false }, function(
-        err,
-        user,
-        info
-    ) {
-        if (err) {
-            return next(err);
-        }
 
-        if (user) {
-            return res.json({ user: user.toAuthJSON() });
-        } else {
-            return res.status(422).json(info);
-        }
-    })(req, res, next);
+    if (user) {
+      const payload = {
+        username: user.username,
+        email: user.email,
+      };
+
+      const token = tokenService.createToken(payload, '3d');
+      return res.status(200).json({
+        data: {
+          email: user.email,
+          username: user.username,
+          token,
+        },
+      });
+    }
+    return res.status(422).json(info);
+  })(req, res, next);
 });
 
-router.post("/users", function(req, res, next) {
-    const username = req.body.user.username,
-    email = req.body.user.email,
-    hash = req.body.user.password;
+router.post('/users', (req, res, next) => {
+  const { username, email, password: hash } = req.body.user;
 
-    User.create({ username, email, hash})
-    .then(function(user){
-        return res.json({
-            email: user.email,
-            username: user.username
-        });
-    }).catch(next);
+  User.create({ username, email, hash })
+    .then((user) => {
+      const payload = {
+        username: user.username,
+        email: user.email,
+      };
 
+      const token = tokenService.createToken(payload, '3d');
```

Comment:
I think I told you something about not directly passing `'3d'` into this call here, define it where you're using it